common/springUtilities/teamFunctions.lua

```diff
@@ -90,6 +93,38 @@ local function getSettings()
 		isCoop = true
 	end
 
+	if Spring.GetModOptions and Spring.GetModOptions().date_day then
+		local currentDay = Spring.GetModOptions().date_day
+		local currentMonth = Spring.GetModOptions().date_month
+		local currentYear = Spring.GetModOptions().date_year
+
+		-- FIXME: This doesn't support events that start and end in different years.
+		for holiday, dates in pairs(holidaysList) do
+			local afterStart = false
+			local beforeEnd = false
+			if (dates.firstDay.month == currentMonth and dates.firstDay.day <= currentDay) or (dates.firstDay.month < currentMonth) then
+				afterStart = true
+			end
+			if (dates.lastDay.month == currentMonth and dates.lastDay.day >= currentDay) or (dates.lastDay.month > currentMonth) then
+				beforeEnd = true
+			end
+
+			if afterStart and beforeEnd then
+				isHoliday[holiday] = true
+				if dates.specialDay and dates.specialDay.day == currentDay and dates.specialDay.month == currentMonth then
+					isHoliday[holiday .. "_specialDay"] = true
+				else
+					isHoliday[holiday .. "_specialDay"] = false
+				end
+			else
+				isHoliday[holiday] = false
+				isHoliday[holiday .. "_specialDay"] = false
```

Comment:
would be nicer if it was pairs of `isHoliday[holiday] = {isWithinHolidayPreiod, isSpecialDayOfThatHoliday}`